server/factory/interface.go

```diff
@@ -9,7 +9,8 @@ import (
 	"github.com/ElrondNetwork/rosetta/server/resources"
 )
 
-type networkProvider interface {
+// NetworkProvider defines the actions that need to be performed by the component that handles network data fetching
+type NetworkProvider interface {
```

Comment:
Here it should have been private (un-exported), though.

Reply:
no, according to the linter, since you return an exported item within an exported function